contracts/thecosomata/ThecosomataETH.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IBTRFLY is IERC20 {
+    function burn(uint256 amount) external;
+
+    function decimals() external view returns (uint8);
+}
+
+interface IRedactedTreasury {
+    function manage(address _token, uint256 _amount) external;
+}
+
+interface ICurveCryptoPool {
+    function add_liquidity(uint256[2] calldata amounts, uint256 min_mint_amount)
+        external
+        payable;
+
+    function calc_token_amount(uint256[2] calldata amounts)
+        external
+        view
+        returns (uint256);
+
+    // Would be replaced by Chainlink based oracle
+    function price_oracle() external view returns (uint256);
+
+    function token() external view returns (address);
+}
+
+contract ThecosomataETH is Ownable {
+    address public immutable BTRFLY;
+    address public immutable WETH;
+    address public immutable CURVEPOOL;
+    address public immutable TREASURY;
+
+    event AddLiquidity(
+        uint256 ethLiquidity,
+        uint256 btrflyLiquidity,
+        uint256 btrflyBurned
+    );
+
+    constructor(
+        address _BTRFLY,
+        address _WETH,
+        address _TREASURY,
+        address _CURVEPOOL
+    ) {
+        require(_BTRFLY != address(0), "Invalid BTRFLY address");
+        BTRFLY = _BTRFLY;
+
+        require(_WETH != address(0), "Invalid WETH address");
+        WETH = _WETH;
+
+        require(_CURVEPOOL != address(0), "Invalid POOL address");
+        CURVEPOOL = _CURVEPOOL;
+
+        require(_TREASURY != address(0), "Invalid TREASURY address");
+        TREASURY = _TREASURY;
+
+        IERC20(_BTRFLY).approve(_CURVEPOOL, 2**256 - 1);
+        IERC20(_WETH).approve(_CURVEPOOL, 2**256 - 1);
+    }
+
+    function checkUpkeep()
```

Comment:
Since we're not going to use Chainlink keepers anymore I think we can remove this. Can you confirm @realkinando ?